python/test/t_FORM_interval.py

```diff
@@ -61,7 +61,10 @@
 
     ot.RandomGenerator.SetSeed(0)
     # algo = getattr(openturns, algoName)(event)
-    algo = ot.FORM(ot.Cobyla(), event, X.getMean())
+    algoOptim = ot.Cobyla()
+    algoOptim.setMaximumConstraintError(1)
```

Comment:
I am sorry, but I do not understand this line. Why disabling the maximum error constraint in this particular unit test?

Reply:
In this test case, I think that for several domains, there is no solution that could reach the limit state. I tried different optimization algorithms (abdorackwitz, nlopt), but noone works.

For example, the fourth problem to be solved is : 

`class=ThresholdEventImplementation antecedent=class=CompositeRandomVector function=class=Function name=Unnamed implementation=class=FunctionImplementation name=Unnamed description=[x0,x1,x2,x3,y0] evaluationImplementation=class=SymbolicEvaluation name=Unnamed inputVariablesNames=[x0,x1,x2,x3] outputVariablesNames=[y0] formulas=[0.0] gradientImplementation=class=SymbolicGradient name=Unnamed evaluation=class=SymbolicEvaluation name=Unnamed inputVariablesNames=[x0,x1,x2,x3] outputVariablesNames=[y0] formulas=[0.0] hessianImplementation=class=SymbolicHessian name=Unnamed evaluation=class=SymbolicEvaluation name=Unnamed inputVariablesNames=[x0,x1,x2,x3] outputVariablesNames=[y0] formulas=[0.0] antecedent=class=UsualRandomVector distribution=class=Normal name=Normal dimension=4 mean=class=Point name=Unnamed dimension=4 values=[0,0,0,0] sigma=class=Point name=Unnamed dimension=4 values=[1,1,1,1] correlationMatrix=class=CorrelationMatrix dimension=4 implementation=class=MatrixImplementation name=Unnamed rows=4 columns=4 values=[1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1] operator=class=Less name=Unnamed threshold=1`

You will see that formula is `0.0` ! So it is non solvable.

Reply:
This unit test is poorly designed for several reasons:
1) There is no particular reason to check a list of intervals by making variations on the finiteness of the bounds. If the algorithm works for one single interval, it will work for any. The only difference is whether the bounds are correctly taken into account when we check that a given point is within the bounds of an interval: this check is done within a unit test of the Interval class.
2) There is no comparison with a known probability value. We only check that the algorithm runs.

Now I understand what you did. Given that the there are unsolvable problems in the list, setting the tolerance to 1 essentially disables the check that you introduced. This is almost equivalent to disabling the whole unit test (but we still check that the constructor builds).

From my point of view, this unit test reveals that the check you introduce is interesting. This is why I validate this PR. There is, however, a work to be done on these unit tests so that relevant tests are chosen and correct reference probability values are used.